tests/maintel/test_mtcs.py

```diff
@@ -1043,6 +1719,230 @@ async def mtdome_tel_azimuth_next(self, *args, **kwargs):
     async def mtdome_tel_light_wind_screen_next(self, *args, **kwargs):
         return self._mtdome_tel_light_wind_screen
 
+    async def mtm1m3_evt_detailed_state(self, *args, **kwargs):
+        await asyncio.sleep(1.0)
+        return self._mtm1m3_evt_detailed_state
+
+    async def mtm1m3_evt_applied_balance_forces(self, *args, **kwargs):
+        await asyncio.sleep(0.25)
+        return self._mtm1m3_evt_applied_balance_forces
+
+    async def mtm1m3_cmd_raise_m1m3(self, *args, **kwargs):
+        raise_m1m3 = kwargs.get("raiseM1M3", True)
+
+        if (
+            raise_m1m3
+            and self._mtm1m3_evt_detailed_state.detailedState
+            == idl.enums.MTM1M3.DetailedState.PARKED
+        ):
+            self._mtm1m3_raise_task = asyncio.create_task(self.execute_raise_m1m3())
+        else:
+            raise RuntimeError(
+                f"MTM1M3 current detailed state is {self._mtm1m3_evt_detailed_state.detailedState!r}."
+            )
+
+    async def mtm1m3_cmd_abort_raise_m1m3(self, *args, **kwargs):
+
+        if self._mtm1m3_evt_detailed_state.detailedState in {
+            idl.enums.MTM1M3.DetailedState.RAISINGENGINEERING,
+            idl.enums.MTM1M3.DetailedState.RAISING,
+        }:
+            self._mtm1m3_abort_raise_task = asyncio.create_task(
+                self.execute_abort_raise_m1m3()
+            )
+        else:
+            raise RuntimeError("M1M3 Not raising. Cannot abort.")
+
+    async def execute_raise_m1m3(self):
+        self.log.debug("Start raising M1M3...")
+        self._mtm1m3_evt_detailed_state.detailedState = (
+            idl.enums.MTM1M3.DetailedState.RAISING
+        )
+        await asyncio.sleep(4.0)
```

Comment:
Is there a way to avoid explicitly sleeping here? Maybe wait for a task or something?

Reply:
Ah this is a task to emulate raising M1M3. I added 4s so it gives me enough time to do stuff while it "executes" (like aborting it) but not long enough that the test will take forever. I think I could have a variable instead of a hard coded value.